com.hazelcast.mapstore.mysql.MySQLGenericMapStoreTest.whenMapStoreDestroy_thenDropMapping [HZ-1695]

_master_ (commit 5bcb6c0fa7d5f3ea4b5f8ad912d87a6be8fb60f8)

Failed on IbmJDK8 in FIPS mode: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8-fips-nightly/451/testReport/com.hazelcast.mapstore.mysql/MySQLGenericMapStoreTest/whenMapStoreDestroy_thenDropMapping/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expecting:
  <[Row{[__map-store.people_o_9fc9eee5_5097_46f0_8f2d_5277831d7749]},
    Row{[__map-store.people_o_e8977381_2561_41fc_ba89_e908002aa76c]}]>
to contain exactly in any order:
  <[Row{[__map-store.people_o_9fc9eee5_5097_46f0_8f2d_5277831d7749]}]>
but the following elements were unexpected:
  <[Row{[__map-store.people_o_e8977381_2561_41fc_ba89_e908002aa76c]}]>

	at com.hazelcast.jet.sql.SqlTestSupport.assertRowsAnyOrder(SqlTestSupport.java:321)
	at com.hazelcast.jet.sql.SqlTestSupport.assertRowsAnyOrder(SqlTestSupport.java:283)
	at com.hazelcast.mapstore.GenericMapStoreTest.lambda$awaitMappingCreated$6(GenericMapStoreTest.java:627)
	at com.hazelcast.mapstore.GenericMapStoreTest$$Lambda$2532/0000000018014930.run(Unknown Source)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1236)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1253)
	at com.hazelcast.mapstore.GenericMapStoreTest.awaitMappingCreated(GenericMapStoreTest.java:626)
	at com.hazelcast.mapstore.GenericMapStoreTest.whenMapStoreDestroy_thenDropMapping(GenericMapStoreTest.java:113)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:822)
```

</details>

<details><summary>Standard output:</summary>

```
07:43:17,166 DEBUG || - [GenericMapStore] hz.friendly_jang.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initializing for map people_o_9fc9eee5_5097_46f0_8f2d_5277831d7749
07:43:22,264  INFO |whenMapStoreDestroy_thenDropMapping| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
07:43:22,264  INFO |whenMapStoreDestroy_thenDropMapping| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
07:43:22,265  INFO |whenMapStoreDestroy_thenDropMapping| - [SimpleTestInClusterSupport] Time-limited test - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
07:43:22,266  INFO |whenMapStoreDestroy_thenDropMapping| - [SimpleTestInClusterSupport] Time-limited test - Destroying 1 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/__sql.catalog]
BuildInfo right after whenMapStoreDestroy_thenDropMapping(com.hazelcast.mapstore.mysql.MySQLGenericMapStoreTest): BuildInfo{version='5.2-SNAPSHOT', build='20220816', buildNumber=20220816, revision=5bcb6c0, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'whenMapStoreDestroy_thenDropMapping(com.hazelcast.mapstore.mysql.MySQLGenericMapStoreTest):'
07:43:15, accumulated pauses: 57 ms, max pause: 11 ms, pauses over 1000 ms: 0
07:43:20, accumulated pauses: 7 ms, max pause: 0 ms, pauses over 1000 ms: 0
```

</details>

Reopening as issue occurred again on: _master_ (commit 85107fc7a2e74924903ec649ad4d74606f76616e)
with IbmJDK8-fips: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8-fips-nightly/472/testReport/com.hazelcast.mapstore.mysql/MySQLGenericMapStoreTest/whenMapStoreDestroyOnNonMaster_thenDropMapping/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expecting:
  <[Row{[__map-store.people_o_ab75d351_c7de_45cd_a61c_7cd60e226686]},
    Row{[__map-store.people_o_c3064fb3_001e_4471_a33e_03f9a6a039c2]}]>
to contain exactly in any order:
  <[Row{[__map-store.people_o_ab75d351_c7de_45cd_a61c_7cd60e226686]}]>
but the following elements were unexpected:
  <[Row{[__map-store.people_o_c3064fb3_001e_4471_a33e_03f9a6a039c2]}]>

	at com.hazelcast.jet.sql.SqlTestSupport.assertRowsAnyOrder(SqlTestSupport.java:322)
	at com.hazelcast.jet.sql.SqlTestSupport.assertRowsAnyOrder(SqlTestSupport.java:284)
	at com.hazelcast.mapstore.GenericMapStoreTest.lambda$awaitMappingCreated$6(GenericMapStoreTest.java:627)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1236)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1253)
	at com.hazelcast.mapstore.GenericMapStoreTest.awaitMappingCreated(GenericMapStoreTest.java:626)
	at com.hazelcast.mapstore.GenericMapStoreTest.whenMapStoreDestroyOnNonMaster_thenDropMapping(GenericMapStoreTest.java:126)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:825)

```
</details>
<details><summary>Standard output:</summary>

```
21:25:26,540 ERROR || - [InitExecutionOperation] ForkJoinPool.commonPool-worker-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] null
com.hazelcast.jet.impl.execution.ExecutionCancellationException: null
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:316) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution0(JobExecutionService.java:719) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.cancelAllExecutions(JobExecutionService.java:249) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.SimpleTestInClusterSupport.supportAfter(SimpleTestInClusterSupport.java:114) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at sun.reflect.GeneratedMethodAccessor25.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55) ~[?:1.8.0]
	at java.lang.reflect.Method.invoke(Method.java:508) ~[?:1.8.0]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:381) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:39) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293) ~[junit-4.13.2.jar:4.13.2]
	at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]
	at java.lang.Thread.run(Thread.java:825) ~[?:2.9 (08-17-2022)]
Caused by: java.util.concurrent.CancellationException
	... 31 more
Finished Running Test: whenSetNonExistingColumnOnSecondMapStore_thenFailToInitialize in 0.125 seconds.
Started Running Test: whenMapStoreDestroyOnNonMaster_thenDropMapping
21:25:26,540 ERROR || - [InitExecutionOperation] ForkJoinPool.commonPool-worker-0 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] null
com.hazelcast.jet.impl.execution.ExecutionCancellationException: null
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:316) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution0(JobExecutionService.java:719) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.cancelAllExecutions(JobExecutionService.java:249) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.SimpleTestInClusterSupport.supportAfter(SimpleTestInClusterSupport.java:114) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at sun.reflect.GeneratedMethodAccessor25.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55) ~[?:1.8.0]
	at java.lang.reflect.Method.invoke(Method.java:508) ~[?:1.8.0]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:381) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:39) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293) ~[junit-4.13.2.jar:4.13.2]
	at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]
	at java.lang.Thread.run(Thread.java:825) ~[?:2.9 (08-17-2022)]
Caused by: java.util.concurrent.CancellationException
	... 31 more
21:25:26,549 ERROR || - [SubmitJobOperation] hz.infallible_hopper.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] null
com.hazelcast.jet.impl.execution.ExecutionCancellationException: null
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:316) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution0(JobExecutionService.java:719) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.cancelAllExecutions(JobExecutionService.java:249) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.SimpleTestInClusterSupport.supportAfter(SimpleTestInClusterSupport.java:114) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at sun.reflect.GeneratedMethodAccessor25.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55) ~[?:1.8.0]
	at java.lang.reflect.Method.invoke(Method.java:508) ~[?:1.8.0]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:381) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:39) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66) ~[hazelcast-5.3.0-SNAPSHOT-tests.jar:5.3.0-SNAPSHOT]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293) ~[junit-4.13.2.jar:4.13.2]
	at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]
	at java.lang.Thread.run(Thread.java:825) ~[?:2.9 (08-17-2022)]
Caused by: java.util.concurrent.CancellationException
	... 31 more
21:25:26,556 DEBUG || - [GenericMapStore] hz.infallible_hopper.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Discovered following mapping columns: id INTEGER, name VARCHAR, age INTEGER
21:25:26,609 DEBUG || - [GenericMapStore] hz.infallible_hopper.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Initializing for map people_o_ab75d351_c7de_45cd_a61c_7cd60e226686
21:25:26,644 ERROR || - [InitExecutionOperation] ForkJoinPool.commonPool-worker-0 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:547) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:328) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:906) ~[?:1.8.0]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:825) ~[?:2.9 (08-17-2022)]
21:25:26,647 ERROR || - [SubmitJobOperation] hz.infallible_hopper.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:547) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:328) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:906) ~[?:1.8.0]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:825) ~[?:2.9 (08-17-2022)]
21:25:31,753  INFO |whenMapStoreDestroyOnNonMaster_thenDropMapping| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
21:25:31,753  INFO |whenMapStoreDestroyOnNonMaster_thenDropMapping| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
21:25:31,754  INFO |whenMapStoreDestroyOnNonMaster_thenDropMapping| - [SimpleTestInClusterSupport] Time-limited test - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
21:25:31,755  INFO |whenMapStoreDestroyOnNonMaster_thenDropMapping| - [SimpleTestInClusterSupport] Time-limited test - Destroying 1 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/__sql.catalog]
BuildInfo right after whenMapStoreDestroyOnNonMaster_thenDropMapping(com.hazelcast.mapstore.mysql.MySQLGenericMapStoreTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20221013', buildNumber=20221013, revision=85107fc, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'whenMapStoreDestroyOnNonMaster_thenDropMapping(com.hazelcast.mapstore.mysql.MySQLGenericMapStoreTest):'
21:25:25, accumulated pauses: 130 ms, max pause: 83 ms, pauses over 1000 ms: 0
21:25:30, accumulated pauses: 10 ms, max pause: 0 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```

</details>

--- a/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
+++ b/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
@@ -290,13 +290,13 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
 
     @Override
     public void destroy() {
-        awaitInitFinished();
+        awaitSuccessfulInit();
         dropMapping(mapping);
     }
 
     @Override
     public GenericRecord load(K key) {
-        awaitInitFinished();
+        awaitSuccessfulInit();
 
         try (SqlResult queryResult = sql.execute(queries.load(), key)) {
             Iterator<SqlRow> it = queryResult.iterator();
@@ -384,7 +384,7 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
      */
     @Override
     public Map<K, GenericRecord> loadAll(Collection<K> keys) {
-        awaitInitFinished();
+        awaitSuccessfulInit();
 
         Object[] keysArray = keys.toArray();
 
@@ -404,7 +404,7 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
 
     @Override
     public Iterable<K> loadAllKeys() {
-        awaitInitFinished();
+        awaitSuccessfulInit();
 
         SqlResult keysResult = sql.execute(queries.loadAllKeys());
 
@@ -419,7 +419,7 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
 
     @Override
     public void store(K key, GenericRecord record) {
-        awaitInitFinished();
+        awaitSuccessfulInit();
 
         int idPos = -1;
         Object[] params = new Object[columnMetadataList.size()];
@@ -502,7 +502,7 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
 
     @Override
     public void storeAll(Map<K, GenericRecord> map) {
-        awaitInitFinished();
+        awaitSuccessfulInit();
 
         for (Entry<K, GenericRecord> entry : map.entrySet()) {
             store(entry.getKey(), entry.getValue());
@@ -511,14 +511,14 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
 
     @Override
     public void delete(K key) {
-        awaitInitFinished();
+        awaitSuccessfulInit();
 
         sql.execute(queries.delete(), key).close();
     }
 
     @Override
     public void deleteAll(Collection<K> keys) {
-        awaitInitFinished();
+        awaitSuccessfulInit();
 
         if (keys.isEmpty()) {
             return;
@@ -532,17 +532,21 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
     }
 
     /**
-     * Awaits successful initialization, if the initialization failed it throws an exception
+     * Awaits successful initialization; if the initialization fails, throws an exception.
      */
+    private void awaitSuccessfulInit() {
+        awaitInitFinished();
+        if (initFailure != null) {
+            throw new HazelcastException("MapStore init failed for map: " + mapName, initFailure);
+        }
+    }
+
     void awaitInitFinished() {
         try {
             boolean finished = initFinished.await(initTimeoutMillis, MILLISECONDS);
             if (!finished) {
-                throw new HazelcastException("MapStore init for map: " + mapName + " timed out after " + initTimeoutMillis
-                        + " ms", initFailure);
-            }
-            if (initFailure != null) {
-                throw new HazelcastException("MapStore init failed for map: " + mapName, initFailure);
+                throw new HazelcastException("MapStore init for map: " + mapName + " timed out after "
+                        + initTimeoutMillis + " ms", initFailure);
             }
         } catch (InterruptedException e) {
             throw new HazelcastException(e);

--- a/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
+++ b/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
@@ -534,7 +534,7 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
     /**
      * Awaits successful initialization, if the initialization failed it throws an exception
      */
-    private void awaitInitFinished() {
+    void awaitInitFinished() {
         try {
             boolean finished = initFinished.await(initTimeoutMillis, MILLISECONDS);
             if (!finished) {

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
@@ -78,10 +78,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         createTable(mapName);
 
         createMapStore();
-
-        assertTrueEventually(() -> {
-            assertRowsAnyOrder(hz, "SHOW MAPPINGS", newArrayList(new Row(MAPPING_PREFIX + mapName)));
-        }, 5);
+        awaitMappingCreated();
     }
 
     @Test
@@ -113,9 +110,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         awaitMappingCreated();
 
         mapStore.destroy();
-        assertTrueEventually(() -> {
-            assertRowsAnyOrder(hz, "SHOW MAPPINGS", newArrayList());
-        }, 5);
+        awaitMappingDestroyed();
     }
 
     @Test
@@ -127,9 +122,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
 
         GenericMapStore<Object> mapStoreNotMaster = createMapStore(instances()[1]);
         mapStoreNotMaster.destroy();
-        assertTrueEventually(() -> {
-            assertRowsAnyOrder(hz, "SHOW MAPPINGS", newArrayList());
-        }, 5);
+        awaitMappingDestroyed();
     }
 
     @Test
@@ -611,6 +604,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
 
         GenericMapStore<K> mapStore = new GenericMapStore<>();
         mapStore.init(instance, properties, mapName);
+        mapStore.awaitInitFinished();
         return mapStore;
     }
 
@@ -628,4 +622,9 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         }, 5);
     }
 
+    private void awaitMappingDestroyed() {
+        assertTrueEventually(() -> {
+            assertRowsAnyOrder(hz, "SHOW MAPPINGS", newArrayList());
+        }, 5);
+    }
 }

